cmd/migrate.go

```diff
@@ -104,11 +110,20 @@ func migrateCmd() *cobra.Command {
 	return migrateCmd
 }
 
-func runRawMigration(ctx context.Context, m *roll.Roll, migration *migrations.RawMigration, complete bool, c *backfill.Config) error {
-	parsedMigration, err := migrations.ParseMigration(migration)
-	if err != nil {
-		return fmt.Errorf("failed to parse migration %q: %w", migration.Name, err)
+// parseMigrations tries to parse all RawMigrations and collects all the errors
+// if any.
+func parseMigrations(migs []*migrations.RawMigration) ([]*migrations.Migration, error) {
+	parsedMigrations := make([]*migrations.Migration, 0, len(migs))
+	var errs error
+	for _, rawMigration := range migs {
+		m, err := migrations.ParseMigration(rawMigration)
+		if err != nil {
+			errs = errors.Join(errs, err)
+		}
+		parsedMigrations = append(parsedMigrations, m)
 	}
-
-	return runMigration(ctx, m, parsedMigration, complete, c)
+	if errs != nil {
+		return nil, fmt.Errorf("incompatible migration(s) found: %w. please update the files.", errs)
```

Comment:
Avoid saying `update the files` here because this assumes the cause of the error. All we know here is that a migration couldn't be parsed; could be a typo.

```suggestion
				return nil, fmt.Errorf("incompatible migration(s): %w", errs)
```